_posts/2024-03-25-Announcing-Data-Prepper-2.7.0.md

```diff
@@ -0,0 +1,120 @@
+
+```
+processor:
+  - grok:
+      match:
+        log: [ "%{COMMONAPACHELOG_DATATYPED}" ]
+
+  - geoip:
+      entries:
+        - source: clientip
+          target: clientlocation
+          include_fields: [latitude, longitude, location, postal_code, country_name, city_name]
+```
+
+When ingesting data using this pipeline, the OpenSearch index will now contain the geolocation fields expressed above, such as the `latitude` and `city_name`.
+Additionally, you can configure template mappings in OpenSearch so that you can display these events in OpenSearch Dashboards using the [Maps feature](https://opensearch.org/docs/latest/dashboards/visualize/maps/).
+
+
+## AWS Secrets Manager support
+
+Now Data Prepper supports [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) as extension plugin applicable to pipeline plugins (i.e. source, buffer, processor, sink).
+Users are allowed to configure the AWS secrets extension through extensions in data-prepper-config.yaml.
+
+Here is an example of how to configure your secrets.
+
+```
+extensions:
+  aws:
+    secrets:
+      host-secret-config:
+        secret_id: <YOUR_SECRET_ID_1>
+        region: <YOUR_REGION_1>
+        sts_role_arn: <YOUR_STS_ROLE_ARN_1>
+        refresh_interval: <YOUR_REFRESH_INTERVAL_1>
+      credential-secret-config:
+        secret_id: <YOUR_SECRET_ID_2>
+        region: <YOUR_REGION_2>
+        sts_role_arn: <YOUR_STS_ROLE_ARN_2>
+        refresh_interval: <YOUR_REFRESH_INTERVAL_2>
+```
+
+Users can also configure this in the `pipeline_configurations` section of a pipeline YAML file.
+
+The `credential-secret-config` is a user-supplied secret configuration Id.
+Pipeline authors can reference secrets within pipeline plugin settings using the pattern of `$aws_secrets:<<my-defined-secret>>``.
+Here is an example of configuring an OpenSearch sink with secret values.
+
+```
+{% raw %}
+source:
+  - opensearch:
+      hosts: [ "${{aws_secrets:host-secret-config}}" ]
+      username: "${{aws_secrets:credential-secret-config:username}}"
+      password: "${{aws_secrets:credential-secret-config:password}}"
+{% endraw %}
+```
+
+In the above example, secrets under `credential-secret-config` is assumed to be stored as the following JSON key-value pairs:
+
+```
+{
+  "username": <YOUR_USERNAME>
+  "password": <YOUR_PASSWORD>
+}
+```
+
+The secret under `host-secret-config` is assumed to be stored as plaintext.
+To support [secret rotation](https://docs.aws.amazon.com/secretsmanager/latest/userguide/rotating-secrets.html) for OpenSearch, the `opensearch` source automatically refreshes its basic credentials, (i.e. username/password) according to the `refresh_interval` by polling the latest secret values.
+
+For more details, please reference our [aws extension plugins](https://opensearch.org/docs/latest/data-prepper/managing-data-prepper/configuring-data-prepper/#aws-extension-plugins) document.
```

Comment:
```suggestion
For more information, please see the [AWS extension plugins](https://opensearch.org/docs/latest/data-prepper/managing-data-prepper/configuring-data-prepper/#aws-extension-plugins) documentation.
```

Reply:
We actually call this extension `aws`. Perhaps we can change this to:

For more information, please see the [`aws` extension](https://opensearch.org/docs/latest/data-prepper/managing-data-prepper/configuring-data-prepper/#aws-extension-plugins) documentation.

Reply:
I think the link is referencing the heading in the documentation.

Reply:
We are calling these "extensions" in Data Prepper, not "extension plugins." The "plugin" part is redundant in both.